tools/global_bss_check.py

```diff
@@ -0,0 +1,216 @@
+
+
+def main():
+    description = "Check that globally visible bss has not been reordered."
+    # TODO
+    epilog = """\
+    N.B. Since this script reads the map files, it can only see globally visible bss; in-function static bss must be examined with other tools.
+    """
+
+    parser = argparse.ArgumentParser(description=description, epilog=epilog, formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("mapFile", help="Path to a map file.")
+    parser.add_argument("mapFileExpected", help="Path to the expected map file. Optional, default is 'expected/mapFile'.", nargs="?", default="")
+    parser.add_argument("-a", "--print-all", help="Print all bss, not just non-matching.", action="store_true")
+    parser.add_argument("-n", "--no-fun-allowed", help="Remove amusing messages.", action="store_true")
+    args = parser.parse_args()
+
+    if args.mapFileExpected == "":
+        args.mapFileExpected = os.path.join("expected", args.mapFile)
+
+    badFiles, missingFiles, comparedDict = compareMapFiles(args.mapFile, args.mapFileExpected)
+    printCsv(badFiles, missingFiles, comparedDict, args.print_all)
+
+    if len(badFiles) + len(missingFiles) != 0:
+        print("", file=os.sys.stderr)
+
+    if len(badFiles) != 0:
+        print(colorama.Fore.RED + "  BAD" + colorama.Style.RESET_ALL)
+
+        for file in badFiles:
+            print(f"bss reordering in {file}", file=os.sys.stderr)
+        print("", file=os.sys.stderr)
+        
+        if not args.no_fun_allowed:
+            print(colorama.Fore.LIGHTWHITE_EX +
+            "  BSS is REORDERED!!\n"
+            "  Oh! MY GOD!!" 
+            + colorama.Style.RESET_ALL, file=os.sys.stderr)
+            print("", file=os.sys.stderr)
+
+    if len(missingFiles) != 0:
+        print(colorama.Fore.YELLOW + "  MISSING" + colorama.Style.RESET_ALL)
+
+        for file in missingFiles:
+            print(f"Symbols missing from {file}", file=os.sys.stderr)
+        print("", file=os.sys.stderr)
+
+        if not args.no_fun_allowed:
+            print(colorama.Fore.LIGHTWHITE_EX + "  Error, should (not) be in here " + colorama.Style.RESET_ALL, file=os.sys.stderr)
+            print("", file=os.sys.stderr)
+
+        print("Some files appear to be missing symbols. Have they been renamed or declared as static? You may need to remake 'expected'", file=os.sys.stderr)
+
+    if len(badFiles) + len(missingFiles) != 0:
+        return 1
+    
+    print("", file=os.sys.stderr)
+    print(colorama.Fore.GREEN + "  GOOD" + colorama.Style.RESET_ALL, file=os.sys.stderr)
+
+    if args.no_fun_allowed:
+        return 0
+
+    print("\n" + colorama.Fore.LIGHTWHITE_EX +
+    colorama.Back.RED + "                                  " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "         CONGRATURATIONS!         " + colorama.Back.RESET + "\n" +
+    colorama.Back.RED + "    All global BSS is correct.    " + colorama.Back.RESET + "\n" +
```

Comment:
This one comes in from the last screen of the built-in crash debugger, which can be accessed by pressing +L, L, Start, R, CR simultaneously when the red underscore is in the top right of the screen from crashing. Speaking of which, there is a capitalization error here that is of the utmost importance to correct.

```suggestion
    colorama.Back.RED + "    All Global BSS is correct.    " + colorama.Back.RESET + "\n" +
```

![debugger](https://user-images.githubusercontent.com/31797144/135630811-403e55b8-86eb-414b-87f2-e570eb517708.PNG)